src/System.Memory/src/System/SpanHelpers.cs

```diff
@@ -106,6 +106,146 @@ private static bool IsReferenceFreeCoreSlow(Type type)
             return true;
         }
 
+        [StructLayout(LayoutKind.Sequential, Size = 64)]
+        struct Reg64 { }
+        [StructLayout(LayoutKind.Sequential, Size = 32)]
+        struct Reg32 { }
+        [StructLayout(LayoutKind.Sequential, Size = 16)]
+        struct Reg16 { }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        unsafe static bool LessThanEqual(this IntPtr index, UIntPtr length)
+        {
```

Comment:
There's a quite a lot of helper code devoted to clearing now - might be nice to segregate those in a SpanHelpers.Clear.cs...